lighthouse-core/gather/gather-runner.js

```diff
@@ -280,9 +279,18 @@ class GatherRunner {
       log.log('status', status);
       const devtoolsLog = driver.endDevtoolsLog();
       const networkRecords = NetworkRecorder.recordsFromLogs(devtoolsLog);
-      GatherRunner.assertPageLoaded(options.url, driver, networkRecords);
       log.verbose('statusEnd', status);
 
+      pageLoadError = GatherRunner.getPageLoadError(options.url, networkRecords);
+      // If the driver was offline, a page load error is expected, so do not save it.
+      if (!driver.online) pageLoadError = null;
```

Comment:
can we do this check within `getPageLoadError`? seems reasonable.

Reply:
heh, I explicitly moved it out of there since it felt like it was in the wrong place over there :) 

in fact I'd eventually want to remove this special check entirely to make the passes on similar footing